lib/kdtree.h

```diff
@@ -247,7 +368,7 @@ class KDTree {
      *    /   \
      * [2 3]  [4 5 6]
      */
-    std::vector<detail::FlatNode> nodes_;
+     const mms::vector<mms::Mmapped, detail::FlatNode>* nodes_;
```

Comment:
Instead we have:

```c++
mms::vector<mms::Mmapped, detail::FlatNode>* load_nodes(std::string path) {
    ...
    const mms::vector<mms::Mmapped, detail::FlatNode>* nodes =
        reinterpret_cast<const mms::vector<mms::Mmapped, detail::FlatNode>*>(data + pos);

    return nodes;
}
...
mms::vector<mms::Mmapped, detail::FlatNode>* nodes
if ("path/to/kdcache") {
    nodes = load_nodes("path/to/kdcache");
} else {
    mms::vector<mms::Standalone, detail::FlatNodeT<Standalone>> nodes_standalone =
        flatten(...);
    nodes = save_nodes(nodes_standalone, "path/to/kdcache")
}

KDTree(nodes)
```

Note that we *have* to save the nodes in order to get `mms::vector<mms::Mmapped, detail::FlatNode>*` type.